solr/src/main/java/org/apache/metamodel/solr/SolrDataContext.java

```diff
@@ -0,0 +1,396 @@
+            }
+        } catch (IOException e) {
+            logger.error("Failed to parse schema", "", e);
+        }
+
+        return tableDef;
+    }
+
+    @Override
+    protected Schema getMainSchema() throws MetaModelException {
+        try {
+            final MutableSchema theSchema = new MutableSchema(
+                    getMainSchemaName());
+
+            final MutableTable table = tableDef.toTable().setSchema(theSchema);
+            theSchema.addTable(table);
+
+            return theSchema;
+        } catch (Exception e) {
+            throw new MetaModelException("Schema retrieval failed " + e);
+        }
+    }
+
+    @Override
+    protected Number executeCountQuery(Table table,
+            List<FilterItem> whereItems, boolean functionApproximationAllowed) {
+        String queryStr = "*:*";
+
+        if (whereItems != null) {
+            for (int i = 0; i < whereItems.size(); i++) {
+                FilterItem whereItem = whereItems.get(i);
+                queryStr = whereItem.toString();
+            }
+        }
+
+        QueryResponse response = selectRows(table, null, queryStr, 0, 0);
+        return response.getResults().getNumFound();
+    }
+
+    @Override
+    public DataSet executeQuery(Query q) throws MetaModelException {
+        String queryStr = "";
+        QueryResponse response = null;
+
+        Schema schema = getMainSchema();
+        Table table = schema.getTable(0);
+        Column[] columns = table.getColumns();
+
+        boolean selectAll = true;
+        List<SelectItem> selectItems = q.getSelectClause().getItems();
+
+        for (SelectItem selectItem : selectItems) {
+            if (selectItem.getFunction() != null
+                    || selectItem.getColumn() == null) {
+                selectAll = false;
+                break;  
+            }
+        }
+
+        if (selectItems.size() == 1 && selectAll == false) {
+            return super.executeQuery(q);
+        }
+        
+        FilterClause clause = q.getWhereClause();
+
+        if (clause != null) {
+            List<FilterItem> filterItemList = clause.getItems();
+
+            boolean isFirst = true;
+            
+            for (FilterItem filterItem : filterItemList) {
+                String expr = filterItem.getExpression();
+
+                if (expr == null){
+                    expr = filterItem.toString();
+                    expr = expr.replaceAll("[a-z_0-9]+\\.","");
+                }
+                
+                expr = expr.replace(" or ", " OR ").replace(" and ", " AND ").replaceAll("=",":");
+                
+                String joiner = (isFirst == true) ? " " : " AND ";
+                queryStr += joiner + expr;
+                
+                isFirst = false;
+            }
+        }
+        
+        if (queryStr.isEmpty()) {
+            queryStr = "*:*";
+        }
+
+        Integer limitObject = q.getMaxRows();
+        int limit = DEFAULT_LIMIT;
```

Comment:
Setting a default limit means that this module behaves differently than other modules. Why is this necesary? It kinda breaks the polymorphic nature of all DataContext implementations - we'd want to facilitate a _write once, run on any datastore_ kind of API.